packages/protocol/test/governance/double_signing_slasher.ts

```diff
@@ -0,0 +1,217 @@
+import { addressToPublicKey } from '@celo/utils/lib/signatureUtils'
+import BigNumber from 'bignumber.js'
+import {
+  AccountsContract,
+  AccountsInstance,
+  MockElectionContract,
+  MockElectionInstance,
+  MockLockedGoldContract,
+  MockLockedGoldInstance,
+  RegistryContract,
+  RegistryInstance,
+  TestDoubleSigningSlasherContract,
+  TestDoubleSigningSlasherInstance,
+  ValidatorsTestContract,
+  ValidatorsTestInstance,
+} from 'types'
+
+const Accounts: AccountsContract = artifacts.require('Accounts')
+const Validators: ValidatorsTestContract = artifacts.require('ValidatorsTest')
+const DoubleSigningSlasher: TestDoubleSigningSlasherContract = artifacts.require(
+  'TestDoubleSigningSlasher'
+)
+const MockElection: MockElectionContract = artifacts.require('MockElection')
+const MockLockedGold: MockLockedGoldContract = artifacts.require('MockLockedGold')
+const Registry: RegistryContract = artifacts.require('Registry')
+
+// @ts-ignore
+// TODO(mcortesi): Use BN
+Validators.numberFormat = 'BigNumber'
+// @ts-ignore
+DoubleSigningSlasher.numberFormat = 'BigNumber'
+
+const HOUR = 60 * 60
+const DAY = 24 * HOUR
+// Hard coded in ganache.
+const EPOCH = 100
+
+contract('DoubleSigningSlasher', (accounts: string[]) => {
+  let accountsInstance: AccountsInstance
+  let validators: ValidatorsTestInstance
+  let registry: RegistryInstance
+  let mockElection: MockElectionInstance
+  let mockLockedGold: MockLockedGoldInstance
+  let slasher: TestDoubleSigningSlasherInstance
+
+  const validatorLockedGoldRequirements = {
+    value: new BigNumber(1000),
+    duration: new BigNumber(60 * DAY),
+  }
+  const groupLockedGoldRequirements = {
+    value: new BigNumber(1000),
+    duration: new BigNumber(100 * DAY),
+  }
+  const validatorScoreParameters = {
+    exponent: new BigNumber(5),
+    adjustmentSpeed: toFixed(0.25),
+  }
+  const membershipHistoryLength = new BigNumber(5)
+  const maxGroupSize = new BigNumber(5)
+
+  // A random 64 byte hex string.
+  const blsPublicKey =
+    '0x4fa3f67fc913878b068d1fa1cdddc54913d3bf988dbe5a36a20fa888f20d4894c408a6773f3d7bde11154f2a3076b700d345a42fd25a0e5e83f4db5586ac7979ac2053cd95d8f2efd3e959571ceccaa743e02cf4be3f5d7aaddb0b06fc9aff00'
+  const blsPoP =
+    '0xcdb77255037eb68897cd487fdd85388cbda448f617f874449d4b11588b0b7ad8ddc20d9bb450b513bb35664ea3923900'
+  const commission = toFixed(1 / 100)
+
+  const registerValidator = async (address: string) => {
+    await mockLockedGold.setAccountTotalLockedGold(address, validatorLockedGoldRequirements.value)
+    const publicKey = await addressToPublicKey(address, web3.eth.sign)
+    await validators.registerValidator(
+      // @ts-ignore bytes type
+      publicKey,
+      // @ts-ignore bytes type
+      blsPublicKey,
+      // @ts-ignore bytes type
+      blsPoP,
+      { from: address }
+    )
+  }
+
+  const registerValidatorGroup = async (group: string) => {
+    await mockLockedGold.setAccountTotalLockedGold(group, groupLockedGoldRequirements.value)
```

Comment:
FYI there's a bug here, we should be locking `groupLockedGoldRequirements.value * numMembers`, see https://github.com/celo-org/celo-monorepo/pull/2216 for a fix